packages/suite/src/views/wallet/send/components/Options/BitcoinOptions/Locktime.tsx

```diff
@@ -24,22 +26,28 @@ export const Locktime = ({ close }: LocktimeProps) => {
         toggleOption,
         formState: { errors },
         composeTransaction,
+        network,
+        watch,
     } = useSendFormContext();
 
+    const blockchain = useSelector(selectNetworkBlockchainInfo(network.symbol));
+
     const { translationString } = useTranslation();
 
     const options = getDefaultValue('options', []);
     const broadcastEnabled = options.includes('broadcast');
     const inputName = 'bitcoinLockTime';
-    const inputValue = getDefaultValue(inputName) || '';
+    const defaultInputValue = getDefaultValue(inputName) || '';
     const error = errors[inputName];
 
-    const handleLocktimeChange = () => {
-        if (!error) {
-            if (broadcastEnabled) toggleOption('broadcast');
+    const locktime = watch('bitcoinLockTime');
+
+    useEffect(() => {
```

Comment:
Does this have to be an effect? I'd prefer it to stay in the change handler, if possible. Effects make the code less predictable.

Reply:
Well, I hate those effects as well.  My understanding of ReactHookForm is that this is the way how to do it and using `onChange` is the bad way (as I cannot guarantee that the form state was already updated). Lets discuss this. I discussed this with @PeKne and we came to this conclusion.

Reply:
I believe it is safe. If you need to access `locktime`, you can do so from `e.target.value`. Not tested.

Reply:
Yes if you need just value from the field itself. But here you also touch `broadcastEnabled` both read & write (in this case it may work, but still, I think we should not do this in general as it can break thinks.

Also calling `composeTransaction` is IMO recipe for a disaster as this function has many™ consequences.

Reply:
I actually think having `composeTransaction` gives you more control over when it is called, which is preferable. `broadcastEnabled ` can also be safely access via `getValues`. Not sure about `getDefaultValue`. But I will try.